File: algorithms/0001.two-sum/solution.go
func twoSum(nums []int, target int) []int {
	m := map[int]int{}
	for idx, n := range nums {
		diff := target - n
		if i, ok := m[diff]; ok {
			return []int{i, idx}
		} else {
			m[n] = idx
		}
	}
	return []int{}
}

File: algorithms/0001.two-sum/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn two_sum(nums: Vec<i32>, target: i32) -> Vec<i32> {
        let mut map = HashMap::with_capacity(nums.len());
        for (idx, n) in nums.iter().enumerate() {
            let diff = target - n;
            if let Some(&i) = map.get(&diff) {
                return vec![i as i32, idx as i32];
            } else {
                map.insert(n, idx);
            }
        }
        vec![]
    }
}
